File: server/src/logger.ts
import { config } from './config';

export const log = {
  debug: (msg: string, ...args: unknown[]) => {
    if (config.debug) console.log(`[debug] ${msg}`, ...args);
  },
  info:  (msg: string, ...args: unknown[]) => console.log(`[info] ${msg}`, ...args),
  warn:  (msg: string, ...args: unknown[]) => console.warn(`[warn] ${msg}`, ...args),
  error: (msg: string, ...args: unknown[]) => console.error(`[error] ${msg}`, ...args),
};

File: c2server/internal/logger.go
package internal

import "log"

func LogDebug(format string, args ...any) {
	if Cfg != nil && Cfg.Debug {
		log.Printf("[debug] "+format, args...)
	}
}

func LogInfo(format string, args ...any) {
	log.Printf("[info] "+format, args...)
}

func LogWarn(format string, args ...any) {
	log.Printf("[warn] "+format, args...)
}

func LogError(format string, args ...any) {
	log.Printf("[error] "+format, args...)
}
